nodes/scene/timer.py

```diff
@@ -0,0 +1,510 @@
+            pause_button.function_name = "pause_timer"
+        else:  # PAUSED
+            start_button = split.operator(cb, text='', icon='PLAY')
+            start_button.function_name = "start_timer"
+
+        # expire_button = split.operator(cb, text='', icon='FF')
+        # expire_button.function_name = "expire_timer"
+
+        row = box.row(align=True)
+        reset_button = row.operator(cb, text='', icon="REW")
+        reset_button.function_name = "reset_timer"
+        row.prop(self, 'timer_slider', text="", slider=True)
+        expire_button = row.operator(cb, text='', icon='FF')
+        expire_button.function_name = "expire_timer"
+
+        layout.separator()
+        col = layout.column(align=True)
+        col.prop(self, "normalize", text="Normalize", toggle=True)
+        layout.separator()
+
+    def draw_buttons_ext(self, context, layout):
+        self.draw_buttons(context, layout)
+        self.draw_animatable_buttons(layout)
+        layout.prop(self, "absolute")
+
+    def start_timer(self, context):
+        print("* Timer: start_timer")
+        for timer in self.timers:
+            timer.start()
+        updateNode(self, context)
+
+    def stop_timer(self, context):
+        print("* Timer: stop_timer")
+        for timer in self.timers:
+            timer.stop()
+        self.sync_time_slider(0)
+        updateNode(self, context)
+
+    def pause_timer(self, context):
+        print("* Timer: pause_timer")
+        for timer in self.timers:
+            timer.pause()
+        updateNode(self, context)
+
+    def reset_timer(self, context):
+        print("* Timer: reset_timer")
+        for timer in self.timers:
+            timer.reset()
+        self.sync_time_slider(0)
+        updateNode(self, context)
+
+    def expire_timer(self, context):
+        print("* Timer: expire_timer")
+        for timer in self.timers:
+            timer.expire()
+
+        self.sync_time_slider(1)
+
+        updateNode(self, context)
+
+    def sv_init(self, context):
+        self.width = 160
+        self.inputs.new('SvStringsSocket', "Duration").prop_name = "timer_duration"
+        self.inputs.new('SvStringsSocket', "Speed").prop_name = "timer_speed"
+        self.inputs.new('SvStringsSocket', "Loops").prop_name = "timer_loops"
+        self.inputs.new('SvStringsSocket', "Operation")
+
+        self.outputs.new('SvStringsSocket', "Status")
+        self.outputs.new('SvStringsSocket', "Elapsed Time")
+        self.outputs.new('SvStringsSocket', "Remaining Time")
+        self.outputs.new('SvStringsSocket', "Expired")
+        self.outputs.new('SvStringsSocket', "Loop")
+
+    def process(self):
+        # print("processing timer node")
+
+        # print("frame: {0} , last frame: {1}, frame diff: {2}, time diff: {3}".format(frame, self.last_frame, frame_diff, time_diff))
+
+        # return if no outputs are connected
+        if not any(s.is_linked for s in self.outputs):
+            return
```

Comment:
Update system itself does not call not connected nodes.

Reply:
@Durman  the update system will call a node that was linked when you un-link it. Because it calls the affected nodes by the link change. this is good (for example) for the viewers to disable the bgl call

Reply:
Btw, I copied the code from other nodes I created in the past and use as a template for new node and always thought this check was a recommended way to do things. If this has changed or never have been the case then let me know and I’ll remove it. I’m not knowledgeable enough the updating processes underneath.

Reply:
it appears that with recent changes to the update system, some nodes that implement `sv_update` may be simplified a little, but if you do remove the `sv_update` then the `process` function may be where you do the extra defensive code.

don't worry about it.

Reply:
Process of nodes looks like does not call if it does not connect with any other nodes. I'm not going find proves of it in code because with it current readability it will probably will take a whole day. But where I'm wrong is that if a node has input connection it will be called by update system and in this case to check linking of output has sense.